### ISSUE TITLE
ArgumentOutOfRangeException when editing code in AGS Editor

### ISSUE THREAD

**Issue by ericoporto** [adventuregamestudio/ags#981]
**Describe the bug**
I started upgrading my project from 3.4.3.1 to 3.5.0 using the newest 3.5.0.19.

The following error pop up occasionally when I select a line of valid code and paste or type something to replace it:

```
Error: Length cannot be less than zero.
Parameter name: length
Version: AGS 3.5.0.19

System.ArgumentOutOfRangeException: Length cannot be less than zero.
Parameter name: length
   at System.String.InternalSubStringWithChecks(Int32 startIndex, Int32 length, Boolean fAlwaysCopy)
   at AGS.Editor.AutoComplete.ConstructCache(Script scriptToCache)
   at AGS.Editor.ScriptEditor.scintilla_OnBeforeShowingAutoComplete(Object sender, EventArgs e)
   at AGS.Editor.ScintillaWrapper.ShowAutoComplete(Int32 charsTyped, String autoCompleteList)
   at AGS.Editor.ScintillaWrapper.ShowAutoCompleteIfAppropriate(Int32 minimumLength)
   at AGS.Editor.ScintillaWrapper.OnUpdateUI(Object sender, EventArgs e)
   at Scintilla.ScintillaControl.DispatchScintillaEvent(SCNotification notification)
   at Scintilla.ScintillaControl.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

I don't know if relevant, but the error pops in a dialog that I can copy the error text, after closing, it pops up again in another dialog, this one I can't edit the text, and after closing this one, it pops up again, after closing for the third time, it's gone.

**AGS Version**
AGS 3.5.0.19, but I had it before on 3.4.3.1, I just never reported because I thought it might have some already fixed bug for some reason.

**Game**
It only happens on my big project, I can send it privately since I have already backed it up.

**To Reproduce**
It doesn't happen always but when it does it goes like this
1. I write some valid piece of code. It's building normally.
2. I select one line of the valid code with the mouse
3. Once the line is blue, I start typing, to replace the line
4. I then get the error

It's rare, I had done the above around 50 times before getting it, when I was editing a script module code.

**Expected behavior**
I just expected to have the text replaced normally.

**Desktop (please complete the following information):**
 - OS: Ubuntu Linux 64-bit, using AGS under Wine 3.20
 - Version 18.04

**Additional context**
I don't seem to get this when I work in smaller projects.

**Comment by ericoporto** [comment]
Ok, I apparently figured. I had a lost comment with `///` somewhere in my scripts like this:
```
function MySay(this Character*, String text) {
 this.Say(text); 
} 

///

```
And after I removed it, the above error stopped randomly triggering.

**Comment by morganwillcock** [comment]
`///` is specifically checked as a special case, to store the text that follows it. It looks like this is because this is how the text descriptions are picked up from the built-in script header. The error just seems to be that the substring call to get the text doesn't account for the chance of there being nothing after the `///`, just adding a newline or a single space after it stops the crash.

So this looks easy to fix, unless there is some other purpose for the special case of `///` that I'm missing.

**Comment by ericoporto** [comment]
This was fixed by #987
